src/rgw/setup/rgw.py

```diff
@@ -558,17 +565,30 @@ def _update_hax_endpoint_and_create_admin(conf: MappedConf):
                         continue
                     status = Rgw._create_rgw_user(conf)
                     if status == 0:
+                        Log.info(f'User creation is successful on "{Rgw._machine_id}" node.')
+                        Rgw._set_consul_kv(rgw_consul_idx, CONSUL_LOCK_KEY, 'user_created')
                         break
                     else:
                         if data_pod_hostname == data_pod_hostnames[-1]:
-                            raise SetupError(status, 'Admin user creation failed ' \
-                                'with all data pods')
+                            Log.error('Admin user creation failed')
```

Comment:
Please check if we have any error message/error code that can be added to log.

Reply:
Added error code in this line,
and also In the create_user function, we are printing the exact error message.

```
_, err, rc, = SimpleProcess(create_usr_cmd).run(timeout=ADMIN_CREATION_TIMEOUT)
        if rc == 0:
            Log.info(f'RGW admin user {user_name} is created.')
            return 0
        elif rc != 0:
            if err and err_str in err.decode():
                Log.info(f'RGW admin user {user_name} is already created. \
                    skipping user creation.')
                return 0
            else:
                Log.error(f'"{create_usr_cmd}" failed with error {err}.')
```